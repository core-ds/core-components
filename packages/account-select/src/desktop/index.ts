export { AccountSelectDesktop } from './Component.desktop';
