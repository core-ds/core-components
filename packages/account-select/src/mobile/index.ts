export { AccountSelectMobile } from './Component.mobile';
