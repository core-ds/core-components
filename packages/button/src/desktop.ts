export { ButtonDesktop } from './Component.desktop';
export type { CommonButtonProps as ButtonDesktopProps } from './typings';
