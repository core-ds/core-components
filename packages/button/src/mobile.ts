export { ButtonMobile } from './Component.mobile';
export type { CommonButtonProps as ButtonMobileProps } from './typings';
