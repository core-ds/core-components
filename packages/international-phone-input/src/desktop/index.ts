export { InternationalPhoneInputDesktop } from './Component.desktop';
export type { InternationalPhoneInputDesktopProps } from '../types';
