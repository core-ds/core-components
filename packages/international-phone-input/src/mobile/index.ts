export { InternationalPhoneInputMobile } from './Component.mobile';
export type { InternationalPhoneInputMobileProps } from '../types';
