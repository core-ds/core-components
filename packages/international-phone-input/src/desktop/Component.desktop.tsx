import React, { forwardRef } from 'react';

import { InputDesktop } from '@alfalab/core-components-input/desktop';
import { InputAutocompleteDesktop } from '@alfalab/core-components-input-autocomplete/desktop';
import { SelectDesktop } from '@alfalab/core-components-select/desktop';

import { BaseInternationalPhoneInput } from '../components/base-international-phone-input';
import type {
    BaseInternationalPhoneInputProps,
    InternationalPhoneInputDesktopProps,
} from '../types';

export const InternationalPhoneInputDesktop = forwardRef<
    HTMLInputElement,
    InternationalPhoneInputDesktopProps
>((props, ref) => (
    <BaseInternationalPhoneInput
        selected={props.value}
        {...props}
        view='desktop'
        ref={ref}
        SelectComponent={SelectDesktop}
        Input={InputDesktop}
        InputAutocomplete={
            InputAutocompleteDesktop as BaseInternationalPhoneInputProps['InputAutocomplete']
        }
    />
));
