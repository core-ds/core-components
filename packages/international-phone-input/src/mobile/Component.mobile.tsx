import React, { forwardRef } from 'react';

import { InputMobile } from '@alfalab/core-components-input/mobile';
import { InputAutocompleteMobile } from '@alfalab/core-components-input-autocomplete/mobile';
import { SelectMobile } from '@alfalab/core-components-select/mobile';

import { BaseInternationalPhoneInput } from '../components/base-international-phone-input';
import type {
    BaseInternationalPhoneInputProps,
    InternationalPhoneInputMobileProps,
} from '../types';

export const InternationalPhoneInputMobile = forwardRef<
    HTMLInputElement,
    InternationalPhoneInputMobileProps
>((props, ref) => (
    <BaseInternationalPhoneInput
        selected={props.value}
        {...props}
        view='mobile'
        ref={ref}
        SelectComponent={SelectMobile}
        Input={InputMobile}
        InputAutocomplete={
            InputAutocompleteMobile as BaseInternationalPhoneInputProps['InputAutocomplete']
        }
    />
));
