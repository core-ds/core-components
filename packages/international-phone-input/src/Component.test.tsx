import React, { useState } from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { InternationalPhoneInputDesktop, InternationalPhoneInputDesktopProps } from './desktop';
import { InternationalPhoneInputMobile } from './mobile';

import {
    getInternationalPhoneInputDesktopTestIds,
    getInternationalPhoneInputMobileTestIds,
} from './utils';

describe('InternationalPhoneInput', () => {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: true,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });

    const InternationalPhoneInputStateful = ({
        onChange,

        ...restProps
    }: InternationalPhoneInputDesktopProps) => {
        const [value, setValue] = useState('');

        const handleChange: InternationalPhoneInputDesktopProps['onChange'] = (phone) => {
            onChange?.(phone);
            setValue(phone);
        };
        return (
            <InternationalPhoneInputDesktop {...restProps} value={value} onChange={handleChange} />
        );
    };

    const testId = 'international-phone-input';

    it('should match snapshot', () => {
        const { container } = render(
            <InternationalPhoneInputDesktop value='+7 111 111 11 11' onChange={jest.fn()} />,
        );

        expect(container).toMatchSnapshot();
    });

    it('should unmount without errors', () => {
        const { unmount } = render(
            <InternationalPhoneInputDesktop value='+7 111 111 11 11' onChange={jest.fn()} />,
        );

        expect(unmount).not.toThrowError();
    });

    it('should call `onChange` callback after input was changed with dial code of country without priority', () => {
        const onChange = jest.fn();
        render(<InternationalPhoneInputDesktop value='' onChange={onChange} dataTestId={testId} />);

        const input = screen.getByDisplayValue('');

        fireEvent.input(input, { target: { value: '+54' } });

        expect(onChange).toHaveBeenCalledWith('+54');
    });

    it.skip('should call `onChange` callback after input was changed with dial code of country from NANP', async () => {
        const onChange = jest.fn();
        render(<InternationalPhoneInputStateful onChange={onChange} />);

        const input = screen.getByDisplayValue('');

        fireEvent.input(input, { target: { value: '+1868' } });

        await waitFor(() => {
            expect(onChange).toHaveBeenCalledWith('+1868');
        });
    });

    it('should call `onChange` callback after input was changed with whole russian number', () => {
        const onChange = jest.fn();
        render(<InternationalPhoneInputStateful onChange={onChange} />);

        const input = screen.getByDisplayValue('');
        fireEvent.input(input, { target: { value: '+74957888878' } });

        expect(onChange).toHaveBeenCalledWith('+7 495 788 88 78');
    });

    it('should have passed country flag icon', () => {
        const { container } = render(
            <InternationalPhoneInputDesktop value='' onChange={jest.fn()} defaultIso2='az' />,
        );

        const flagComponent = container.querySelector('.flagIcon');

        expect(flagComponent).toHaveAttribute('data-test-id', 'flag-icon-az');
    });

    it.skip('should set new country flag icon from props', async () => {
        const { container } = render(<InternationalPhoneInputDesktop value='+61' />);

        await waitFor(() =>
            expect(container.querySelector('.flagIcon')).toHaveAttribute(
                'data-test-id',
                'flag-icon-au',
            ),
        );
    });

    it('should call `onChange` callback after select was changed', async () => {
        const onChange = jest.fn();

        const { container, getAllByRole } = render(
            <InternationalPhoneInputDesktop value='' onChange={onChange} />,
        );

        const flagComponent = container.querySelector('span[class^="flagIcon"]');

        fireEvent.click(flagComponent as HTMLSpanElement);
        const option = await waitFor(() => getAllByRole('option')[0]);
        fireEvent.click(option);

        expect(onChange).toHaveBeenCalled();
    });

    it('should focus on input after select was changed', async () => {
        const { container, getAllByRole } = render(
            <InternationalPhoneInputDesktop value='+7' onChange={() => null} dataTestId={testId} />,
        );
        const input = await screen.findByDisplayValue('+7');
        const flagComponent = container.querySelector('span[class^="flagIcon"]');

        fireEvent.click(flagComponent as HTMLSpanElement);
        const option = await waitFor(() => getAllByRole('option')[0]);
        fireEvent.click(option);

        await waitFor(() => {
            expect(document.activeElement).toBe(input);
        });
    });

    it('should call `onCountryChange` callback after country was changed', async () => {
        const onCountryChange = jest.fn();
        const { container, getAllByRole } = render(
            <InternationalPhoneInputDesktop
                value=''
                onChange={() => null}
                onCountryChange={onCountryChange}
                dataTestId={testId}
            />,
        );

        const flagComponent = container.querySelector('span[class^=flagIcon]');
        fireEvent.click(flagComponent as HTMLSpanElement);
        const option = await waitFor(() => getAllByRole('option')[0]);
        fireEvent.click(option);

        await waitFor(() => expect(onCountryChange).toHaveBeenCalledTimes(1));
    });

    it('should call `onCountryChange` callback after input was changed', async () => {
        const onCountryChange = jest.fn();
        render(
            <InternationalPhoneInputDesktop
                value=''
                onChange={() => null}
                dataTestId={testId}
                onCountryChange={onCountryChange}
            />,
        );
        const input = screen.getByDisplayValue('');

        fireEvent.input(input, { target: { value: '+998 12 345 67 89', selectionStart: 0 } });

        expect(onCountryChange).toBeCalledWith(
            expect.objectContaining({
                iso2: 'uz',
            }),
        );
    });

    it.skip('should show a blank icon if the country code is not defined after changing the input', async () => {
        const onCountryChange = jest.fn();
        const { container } = render(
            <InternationalPhoneInputDesktop
                value=''
                onChange={() => null}
                dataTestId={testId}
                onCountryChange={onCountryChange}
                defaultIso2='az'
            />,
        );
        const input = screen.getByDisplayValue('');

        fireEvent.input(input, { target: { value: '+000 12 345 67 89', selectionStart: 0 } });

        const emptyCountryIcon = container.querySelector('.emptyCountryIcon');
        expect(emptyCountryIcon).toBeInTheDocument();
    });

    it('should clear the value and set the default flag', async () => {
        const onChange = jest.fn();
        const { container } = render(
            <InternationalPhoneInputDesktop
                value='+7'
                defaultIso2='az'
                onChange={onChange}
                clear={true}
                clearableCountryCode={true}
            />,
        );

        const clearButton = await screen.findByLabelText('Очистить');
        expect(clearButton).toBeInTheDocument();

        fireEvent.click(clearButton);

        const flagComponent = container.querySelector('.flagIcon');
        expect(flagComponent).toHaveAttribute('data-test-id', 'flag-icon-az');
    });

    it('should clear the value and set an empty icon', async () => {
        const onChange = jest.fn();
        const { container } = render(
            <InternationalPhoneInputDesktop
                value='+7'
                clearableCountryCode={true}
                onChange={onChange}
                onCountryChange={() => undefined}
                clear={true}
            />,
        );

        const clearButton = await screen.findByLabelText('Очистить');
        expect(clearButton).toBeInTheDocument();

        fireEvent.click(clearButton);

        const emptyCountryIcon = container.querySelector('.emptyCountryIcon');
        expect(emptyCountryIcon).toBeInTheDocument();
    });

    it('should remove country code', async () => {
        const onChange = jest.fn();
        render(<InternationalPhoneInputDesktop value='+7' onChange={onChange} />);

        const input = await screen.findByDisplayValue('+7');
        await userEvent.type(input, '{backspace}');

        await waitFor(() => {
            expect(onChange).toHaveBeenCalledWith('');
        });
    });

    it('should not remove country code', async () => {
        const onChange = jest.fn();
        render(
            <InternationalPhoneInputDesktop
                value='+7'
                onChange={onChange}
                clearableCountryCode={false}
            />,
        );

        const input = await screen.findByDisplayValue('+7');
        fireEvent.input(input, { target: { value: '+' } });

        await waitFor(() => {
            expect(input).toHaveValue('+7 ');
        });
    });

    it('should not show country select when one country is available', async () => {
        const countries = ['ru'];
        const { queryByTestId } = render(
            <InternationalPhoneInputDesktop
                value='+7'
                countries={countries}
                onChange={() => null}
            />,
        );

        const countrySelect = queryByTestId('countries-select');

        expect(countrySelect).toBeNull();
    });

    it('should show empty country', async () => {
        const onCountryChange = jest.fn();
        render(
            <InternationalPhoneInputDesktop
                value=''
                onChange={() => null}
                dataTestId={testId}
                onCountryChange={onCountryChange}
            />,
        );

        const icons = screen.getAllByRole('img');

        expect(icons[0]).toHaveClass('emptyCountryIcon');
    });

    it('should not render country select when use `hideCountrySelect`', () => {
        const onChange = jest.fn();
        render(
            <InternationalPhoneInputDesktop
                value=''
                onChange={onChange}
                dataTestId={testId}
                countrySelectProps={{ hideCountrySelect: true }}
                defaultIso2='ru'
            />,
        );

        const countrySelect = screen.queryByTestId('countries-select');

        expect(countrySelect).toBeNull();
    });

    it('should clear value', async () => {
        const onChange = jest.fn();
        render(<InternationalPhoneInputDesktop value='+7' onChange={onChange} clear={true} />);

        const clearButton = await screen.findByLabelText('Очистить');

        expect(clearButton).toBeInTheDocument();

        fireEvent.click(clearButton);

        await waitFor(() => {
            expect(onChange).toBeCalledWith('');
        });
    });

    it('should format number when set unformated value', async () => {
        const onChange = jest.fn();
        const onCountryChange = jest.fn();
        const RenderComponent = () => {
            const [value, setValue] = useState('+7 789 123 45 67');

            return (
                <React.Fragment>
                    <InternationalPhoneInputDesktop
                        value={value}
                        onCountryChange={onCountryChange}
                        onChange={(phone) => {
                            setValue(phone);
                            onChange(phone);
                        }}
                    />
                    <button
                        type='button'
                        data-test-id='intl-change-number'
                        onClick={() => setValue('+79491234567')}
                    >
                        Изменить номер
                    </button>
                </React.Fragment>
            );
        };
        render(<RenderComponent />);

        const btn = await screen.findByTestId('intl-change-number');

        fireEvent.click(btn);

        await waitFor(async () => {
            expect(onChange).toHaveBeenCalledWith('+7 949 123 45 67');
            expect(onCountryChange).toHaveBeenCalledWith(
                expect.objectContaining({
                    iso2: 'ru',
                }),
            );
        });
    });

    it('should be not clearable country code', async () => {
        const onChange = jest.fn();
        render(
            <InternationalPhoneInputDesktop
                clearableCountryCode={false}
                value='+7 928 123 45 67'
                onChange={onChange}
                defaultIso2='ru'
            />,
        );

        const input = screen.getByDisplayValue('+7 928 123 45 67');

        for (let i = 0; i < 20; i++) {
            await userEvent.type(input, '{backspace}');
        }

        expect(onChange).toHaveBeenCalledWith('+7');
    });

    it('should be remove chars in selection', async () => {
        const testId = 'input';
        const onChange = jest.fn();
        const { getByTestId } = render(
            <InternationalPhoneInputDesktop
                dataTestId={testId}
                value='+7 983 123 45 67'
                onChange={onChange}
            />,
        );

        const input = getByTestId(testId) as HTMLInputElement;

        await userEvent.type(input, '{backspace}', {
            initialSelectionEnd: 13,
            initialSelectionStart: 10,
        });

        await waitFor(() => {
            expect(onChange).toBeCalledWith('+7 983 123 67');
        });
    });

    it('should not be call onChange', async () => {
        const testId = 'input';
        const onChange = jest.fn();
        const { getByTestId } = render(
            <InternationalPhoneInputDesktop
                dataTestId={testId}
                value='+7 983 123 45 67'
                onChange={onChange}
            />,
        );

        const input = getByTestId(testId) as HTMLInputElement;

        await userEvent.type(input, 'f', {
            initialSelectionEnd: 10,
            initialSelectionStart: 10,
        });

        await waitFor(() => {
            expect(onChange).not.toBeCalled();
        });
    });

    it('should set `data-test-id` attribute in mobile component autocomplete', () => {
        const dti = 'test-dti';
        const { getByTestId } = render(
            <InternationalPhoneInputMobile
                options={[{ key: '+7 921 681 53 98' }]}
                open={true}
                fieldProps={{
                    rightAddons: 'right',
                    error: 'error',
                }}
                inputProps={{
                    error: 'error',
                    rightAddons: 'right',
                }}
                dataTestId={dti}
            />,
        );

        const testIds = getInternationalPhoneInputMobileTestIds(dti);

        expect(getByTestId(testIds.optionsList)).toBeInTheDocument();
        expect(getByTestId(testIds.option)).toBeInTheDocument();
        expect(getByTestId(testIds.bottomSheet)).toBeInTheDocument();
        expect(getByTestId(testIds.bottomSheetHeader)).toBeInTheDocument();
        expect(getByTestId(testIds.bottomSheetContent)).toBeInTheDocument();
        expect(getByTestId(testIds.clearButton)).toBeInTheDocument();
        expect(getByTestId(testIds.applyButton)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteWrapper)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteInner)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteFormControl)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteRightAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteLeftAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteError)).toBeInTheDocument();
        expect(getByTestId(testIds.searchInput)).toBeInTheDocument();
        expect(getByTestId(testIds.searchFormControl)).toBeInTheDocument();
        expect(getByTestId(testIds.searchInner)).toBeInTheDocument();
        expect(getByTestId(testIds.searchLeftAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.searchRightAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.searchError)).toBeInTheDocument();

        const { getByTestId: getByTestIdHint } = render(
            <InternationalPhoneInputMobile
                options={[{ key: '+7 921 681 53 98' }]}
                open={true}
                fieldProps={{ hint: 'hint' }}
                inputProps={{ hint: 'hint' }}
                dataTestId={dti}
            />,
        );

        expect(getByTestIdHint(testIds.fieldAutocompleteHint)).toBeInTheDocument();
        expect(getByTestIdHint(testIds.searchHint)).toBeInTheDocument();
    });

    it('should set `data-test-id` attribute in mobile component country ', () => {
        const dti = 'test-dti';
        const { getByTestId, getAllByTestId } = render(
            <InternationalPhoneInputMobile countrySelectProps={{ open: true }} dataTestId={dti} />,
        );

        const testIds = getInternationalPhoneInputMobileTestIds(dti);

        expect(getAllByTestId(testIds.countryOption).length).toBe(16);
        expect(getByTestId(testIds.countryOptionsList)).toBeInTheDocument();
        expect(getByTestId(testIds.countryBottomSheet)).toBeInTheDocument();
        expect(getByTestId(testIds.countryBottomSheetHeader)).toBeInTheDocument();
        expect(getByTestId(testIds.countryBottomSheetContent)).toBeInTheDocument();
    });

    it('should set `data-test-id` attribute in mobile component no autocomplete', () => {
        const dti = 'test-dti';
        const { getByTestId } = render(
            <InternationalPhoneInputMobile
                options={undefined}
                error='error'
                rightAddons='right'
                dataTestId={dti}
            />,
        );

        const testIds = getInternationalPhoneInputMobileTestIds(dti);

        expect(getByTestId(testIds.field)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldInner)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldFormControl)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldRightAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldLeftAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldError)).toBeInTheDocument();

        const { getByTestId: getByTestIdHint } = render(
            <InternationalPhoneInputMobile options={undefined} hint='hint' dataTestId={dti} />,
        );

        expect(getByTestIdHint(testIds.fieldHint)).toBeInTheDocument();
    });

    it('should set `data-test-id` attribute in desktop component autocomplete', () => {
        const dti = 'test-dti';

        const { getByTestId, container } = render(
            <InternationalPhoneInputDesktop
                options={[{ key: '+7 921 681 53 98' }]}
                open={true}
                error='error'
                inputProps={{
                    rightAddons: 'right',
                }}
                dataTestId={dti}
            />,
        );

        const testIds = getInternationalPhoneInputDesktopTestIds(dti);

        expect(getByTestId(testIds.optionsList)).toBeInTheDocument();
        expect(getByTestId(testIds.option)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteWrapper)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteInner)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteFormControl)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteLeftAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteRightAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldAutocompleteError)).toBeInTheDocument();

        const { getByTestId: getByTestIdHint } = render(
            <InternationalPhoneInputDesktop
                options={[{ key: '+7 921 681 53 98' }]}
                hint='hint'
                dataTestId={dti}
            />,
        );

        expect(getByTestIdHint(testIds.fieldAutocompleteHint)).toBeInTheDocument();
    });

    it('should set `data-test-id` attribute in desktop component country ', () => {
        const dti = 'test-dti';
        const { getByTestId, getAllByTestId } = render(
            <InternationalPhoneInputDesktop countrySelectProps={{ open: true }} dataTestId={dti} />,
        );

        const testIds = getInternationalPhoneInputDesktopTestIds(dti);

        expect(getAllByTestId(testIds.countryOption).length).toBe(16);
        expect(getByTestId(testIds.countryOptionsList)).toBeInTheDocument();
    });

    it('should set `data-test-id` attribute in desktop component no autocomplete', () => {
        const dti = 'test-dti';

        const { getByTestId, container } = render(
            <InternationalPhoneInputDesktop
                options={undefined}
                error='error'
                rightAddons='right'
                dataTestId={dti}
            />,
        );

        const testIds = getInternationalPhoneInputDesktopTestIds(dti);

        expect(getByTestId(testIds.field)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldInner)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldFormControl)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldLeftAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldRightAddons)).toBeInTheDocument();
        expect(getByTestId(testIds.fieldError)).toBeInTheDocument();

        const { getByTestId: getByTestIdHint } = render(
            <InternationalPhoneInputDesktop options={undefined} hint='hint' dataTestId={dti} />,
        );

        expect(getByTestIdHint(testIds.fieldHint)).toBeInTheDocument();
    });
});
