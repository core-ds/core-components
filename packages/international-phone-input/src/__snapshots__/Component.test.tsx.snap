// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternationalPhoneInput should match snapshot 1`] = `
<div>
  <div
    class="component component component size-56 size-56 hasLeftAddons"
  >
    <div
      class="inner inner inner inner filled"
    >
      <div
        class="addons leftAddons addons"
      >
        <div
          class="component"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="component"
            role="combobox"
            tabindex="-1"
          >
            <div
              class="component size-48"
            >
              <div
                aria-controls="downshift-:r0:-menu"
                aria-labelledby="downshift-:r0:-label"
                class="inner"
                id="downshift-:r0:-input"
                tabindex="0"
              >
                <span
                  class="flagIconContainer"
                >
                  <span
                    class="flagIcon"
                    data-test-id="flag-icon-ru"
                  >
                    <svg
                      fill="none"
                      height="24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                      />
                      <path
                        d="M22.2 4.1H1.8C.8 4.1 0 5 0 6v3.5h24V5.9q-.2-1.7-1.8-1.8"
                        fill="#F5F5F5"
                      />
                      <path
                        d="M0 18c0 1 .8 2 1.8 2h20.4q1.6-.2 1.8-1.8v-3.5H0v3.5Z"
                        fill="#FF4B55"
                      />
                      <path
                        d="M24 9.4H0v5.2h24z"
                        fill="#41479B"
                      />
                    </svg>
                  </span>
                </span>
                <svg
                  class="arrow"
                  fill="currentColor"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="inputWrapper"
      >
        <div
          class="input"
        >
          <input
            class="input input"
            type="tel"
            value="+7 111 111 11 11"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
